### PR TITLE
[Bug] File.exists return true when agent.exe doesn't exist but agent folder exist.

### PR DESCRIPTION
**Pull request by Pythyu**
### What does this PR do?

Fix a bug where `File.exists` return true when `agent.exe` doesn't exist but `agent` folder exist. Instead we're using `File.file` that checks if it's a file and if it exist.

### CODE REVIEW

**Review by chouquette — APPROVED** [review]
LGTM!

File: libraries/recipe_helpers.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
-#
+
```
Comment (amenasria) [inline libraries/recipe_helpers.rb]:
```suggestion
#
```

File: libraries/recipe_helpers.rb [hunk 2]
```diff
@@ -3,7 +3,6 @@
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
-#
```
Comment (amenasria) [inline libraries/recipe_helpers.rb:6]: **🥜 nitpick:** ‏Think we shouldn't modify this

File: libraries/recipe_helpers.rb [hunk 3]
```diff
@@ -195,15 +194,15 @@ def must_reinstall?(node)
...
+          return nil unless File.file?(WIN_BIN_PATH)
           shell_out("\"#{WIN_BIN_PATH}\" version -n").stdout.strip
         end
 
         def fetch_current_version
           raw_version = agent_get_version
           return nil if raw_version.nil?
           match_data = raw_version.match(/^Agent ([^\s]*) (- Meta: ([^\s]*) )?- Commit/)
-
+          # will fail if raw_version is empty (Line 199 fails)
```
Comment (amenasria) [inline libraries/recipe_helpers.rb:205]: **💬 suggestion:** ‏We shouldn't mention a line as the file keeps changing !

**Review by amenasria — APPROVED** [review]
Small nits otherwise LGTM!

File: libraries/recipe_helpers.rb (hunk 3 above)
Comment (Pythyu) [reply libraries/recipe_helpers.rb:205]: will edit the comment 👍

**Pull request merged**